packages/core-blockchain/lib/state-storage.js

```diff
@@ -108,7 +108,7 @@ class StateStorage {
    */
   getLastBlocksByHeight (start, end) {
     end = end || start
-    return _mapToBlockData(_lastBlocks.filter(block => block.data.height >= start && block.data.height <= end)).toArray()
+    return _mapToBlockData(_lastBlocks.filter(block => block.data.height >= start && block.data.height <= end)).valueSeq().toArray()
```

Comment:
I would keep logic of arrays. Where the length of returned array would be end-start.
Here it is end-start+1. 

With default end = start+1

And filter with < end

Reply:
Will make that change in another PR. The change from `toArray()` to `valueSeq().toArray()` is just because of a breaking change in the `immutable.js` package.